changelog.d/1635.change.rst

```diff
@@ -0,0 +1 @@
+Resource paths are passed to ``pkg_resources.resource_string`` and similar no longer accept paths that traverse parents or begin with a leading ``/``. Violations of this expectation raise DeprecationWarnings and will become errors.
```

Comment:
I'd change that to mention that absolute path are not supported, rather than saying `/` is not allowed:
```
Resource paths passed to ``pkg_resources.resource_string`` and similar no longer accept paths that traverse parents or are absolute. Violations of this expectation raise DeprecationWarnings and will become errors. Use of Windows path separators (``\``) is no longer allowed.
```